### PR TITLE
Cap passive research points so people spend points more

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
This is not really a 'nerf', but it encourages people to check what techs are available rather than going AFK and unlocking them all at once. Should be a very small change with decent impacts on how invested people are in science. It's based on the count of point sources because my research design lets people build more, although it's less relevant here.

You can bikeshed the 30,000 a bit, the intent is just to avoid people going afk for 45 minutes and coming back to 2 million points

**Changelog**

:cl: Rane
- tweak: Passive research points will not generate above 30,000 stored points. Rate or total available research is not affected. All this means is you have to actively spend them, rather than let them accumulate the whole round and unlock all techs at once.

### CODE REVIEW

**Comment by Elijahrane** [comment]
hang on after banging my head with electro for a bit I found a small bug here I'll fix tomorrow, the map change will be unneccesary

**Comment by Elijahrane** [comment]
Ok ready for review

File: Content.Server/Research/Systems/ResearchSystem.Server.cs [hunk 1]
```diff
@@ -96,7 +102,8 @@ public int PointsPerSecond(ResearchServerComponent component)
     {
         var points = 0;
 
-        if (CanRun(component))
+        // Is our machine powered, and are we below our limit of passive point gain?
+        if (CanRun(component) && component.Points < (component.PassiveLimitPerSource * component.PointSources.Count))
```
Comment (0x6273) [inline Content.Server/Research/Systems/ResearchSystem.Server.cs:106]: With this logic it's possible for points to go over the limit slightly if there are enough point sources to take it from <30k to >30k in one second.

Comment (Elijahrane) [reply Content.Server/Research/Systems/ResearchSystem.Server.cs:106]: doesn't really effect anything, the techs are all multiples of 1000. It's also intended that you can go over with research disks, too

Comment (0x6273) [reply Content.Server/Research/Systems/ResearchSystem.Server.cs:106]: Yeah, this bug is only possible when there is more than one point source, and I'm pretty sure sci research is getting reworked anyway which will probably get rid of point sources entirely.

Comment (Elijahrane) [reply Content.Server/Research/Systems/ResearchSystem.Server.cs:106]: it happens live when centcomm loads

File: Content.Server/Research/Components/ResearchServerComponent.cs [hunk 2]
```diff
@@ -11,8 +11,13 @@ public sealed class ResearchServerComponent : Component
...
+        /// <summary>
+        /// To encourage people to spend points,
+        /// will not accept passive points gain above this number for each source.
+        /// </summary>
+        [DataField("passiveLimitPerSource")]
+        public int PassiveLimitPerSource = 30000;
         [ViewVariables(VVAccess.ReadOnly)] public int Id { get; set; }
-
         [ViewVariables(VVAccess.ReadOnly)]
         public List<ResearchPointSourceComponent> PointSources { get; } = new();
```
Comment (0x6273) [inline Content.Server/Research/Components/ResearchServerComponent.cs:22]: Can you please put newlines between properties?

**Review by 0x6273 — CHANGES_REQUESTED** [review]

**Review by 0x6273 — APPROVED** [review]

**Comment by mirrorcult** [comment]
i think this has potential to be a little unintuitive but people can deal

**Pull request merged**